### PR TITLE
Correct Kind setting for MultiClusterController

### PR DESCRIPTION
**Pull request by vincent-pli**
**What this PR does / why we need it**:
The original implements can get the `Kind` when creating `MultiClusterController`, the `GroupVersionKind()` is empty.

### CODE REVIEW

**Comment by Fei-Guo** [comment]
/hold

Let me double check if this works for CRDs.

File: virtualcluster/pkg/util/mccontroller/mccontroller.go [hunk 1]
```diff
@@ -114,13 +115,17 @@ type ClusterInterface interface {
 
 // NewMCController creates a new MultiClusterController.
 func NewMCController(objectType client.Object, objectListType client.ObjectList, rc reconciler.DWReconciler, opts ...OptConfig) (*MultiClusterController, error) {
-	kind := objectType.GetObjectKind().GroupVersionKind().Kind
+	kinds, _, err := scheme.Scheme.ObjectKinds(objectType)
```
Comment (christopherhein) [inline virtualcluster/pkg/util/mccontroller/mccontroller.go:118]: I believe you should be using this `scheme.Scheme` global variable from - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/virtualcluster/pkg/syncer/util/scheme/scheme.go#L21 which is what other packages reference, this should allow a non-core registered type like a CRD to work.

Comment (vincent-pli) [reply virtualcluster/pkg/util/mccontroller/mccontroller.go:118]: @christopherhein 
I believe you are right, the `scheme` from `controller-runtime` or `client-go` do not include CRD's scheme,
thanks

File: virtualcluster/pkg/util/mccontroller/mccontroller.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
```
Comment (christopherhein) [inline virtualcluster/pkg/util/mccontroller/mccontroller.go:34]:
```suggestion
	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/util/scheme"
```

**Review by christopherhein — CHANGES_REQUESTED** [review]

**Review by christopherhein — APPROVED** [review]
/lgtm

@Fei-Guo did you check this out?

**Pull request merged**